File: backend/src/main/java/com/amzexpress/entity/Address.java
package com.amzexpress.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.time.LocalDateTime;

@Entity
@Table(name = "addresses")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;
    
    @Column(name = "address_type")
    private String addressType; // 'SHIPPING' or 'BILLING'
    
    @Column(name = "street", nullable = false)
    private String street;
    
    @Column(name = "city")
    private String city;
    
    @Column(name = "state")
    private String state;
    
    @Column(name = "postal_code")
    private String postalCode;
    
    @Column(name = "country")
    private String country;
    
    @Column(name = "phone")
    private String phone;
    
    @Column(name = "is_default")
    private Boolean isDefault;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (isDefault == null) isDefault = false;
        if (country == null) country = "Ecuador";
    }
}

File: frontend/src/services/address.ts
import axios from './api'

export interface Address {
  id?: number
  addressType: 'SHIPPING' | 'BILLING'
  street: string
  city: string
  state: string
  postalCode: string
  country: string
  phone: string
  isDefault: boolean
}

export const addressService = {
  async getAll(): Promise<Address[]> {
    const response = await axios.get('/addresses')
    return response.data
  },

  async getShipping(): Promise<Address[]> {
    const response = await axios.get('/addresses/shipping')
    return response.data
  },

  async getBilling(): Promise<Address[]> {
    const response = await axios.get('/addresses/billing')
    return response.data
  },

  async getDefaultShipping(): Promise<Address | null> {
    try {
      const response = await axios.get('/addresses/default-shipping')
      return response.data
    } catch {
      return null
    }
  },

  async create(address: Omit<Address, 'id'>): Promise<Address> {
    const response = await axios.post('/addresses', address)
    return response.data
  },

  async update(id: number, address: Omit<Address, 'id'>): Promise<Address> {
    const response = await axios.put(`/addresses/${id}`, address)
    return response.data
  },

  async delete(id: number): Promise<void> {
    await axios.delete(`/addresses/${id}`)
  }
}